rabix-backend-local/src/main/java/org/rabix/backend/local/BackendCommandLine.java

```diff
@@ -323,4 +405,30 @@ private static Resources extractResources(Map<String, Object> inputs, ProtocolTy
     }
   }
 
+  private static Object createInputValue(String[] value, DataType inputType) {
+    if (value.length > 1 || inputType.isArray()) {
+      if (inputType.isFile()) {
+        List<Map<String, Object>> ret = new ArrayList<>();
+        for (String s : value) {
+          Map<String, Object> entry = new HashMap<>();
+          entry.put("class", "File");
+          entry.put("path", s);
+          ret.add(entry);
+        }
+        return ret;
+      } else {
+        return Arrays.asList(value);
```

Comment:
Logic for creating File and Directory values needs to be encapsulated into the concrete Bindings implementation.